> Looks like the newly added test is failing,
> 
> ```
> ERROR: testGetVariableByName (__main__.GetDependentVariablesTest)
> GetDependentVariablesTest.testGetVariableByName
> ----------------------------------------------------------------------
> Traceback (most recent call last):
>   File "[...]tensorflow/python/ops/gradients_test.py", line 1033, in testGetVariableByName
>     var2 = custom_gradient.get_variable_by_name("a")
>   File "[...]tensorflow/python/ops/custom_gradient.py", line 330, in get_variable_by_name
>     raise ValueError("Unsuccessful at finding variable {}.".format(var_name))
> ValueError: Unsuccessful at finding variable a.
> ```
> 
> It's [tagged not to run in the outside-Google presubmits](https://github.com/tensorflow/tensorflow/blob/371d215a7b40bc083f593e353f775ae71d73f2c4/tensorflow/python/BUILD#L2699) because it was flaky on GPU a few years ago... you may want to run it manually for now, or try removing the tag.

It passes the test again.

--- a/tensorflow/python/ops/gradients_test.py
+++ b/tensorflow/python/ops/gradients_test.py
@@ -1028,7 +1028,10 @@ class GetDependentVariablesTest(test_util.TensorFlowTestCase):
     with context.graph_mode():
       init = constant_op.constant(100.0)
       var = variable_scope.variable(init, name='a/replica_1')
-      var._handle = array_ops.identity(var, name='a')
+      if isinstance(var, variables.RefVariable):
+        var._variable = array_ops.identity(var, name='a')
+      else:
+        var._handle = array_ops.identity(var, name='a')
       var2 = custom_gradient.get_variable_by_name('a')
       self.assertEqual(var2.name, var.name)
   

